blog/_posts/2016-06-12-offset-arrays.md

```diff
@@ -0,0 +1,246 @@
+---
+layout: post
+title:  "New array abstractions for julia-0.5"
+author: <a href="http://holylab.wustl.edu">Tim Holy</a>
+---
+
+# Introduction
+
+Arrays with arbitrary indices are coming as an experimental feature to
+julia-0.5. Conventionally, Julia's arrays are indexed starting at 1,
+whereas some other languages start numbering at 0, and yet others
+(e.g., Fortran) allow you to specify arbitrary starting indices. While
+there is much merit in picking a standard (i.e., 1 for Julia), there
+are some algorithms which simplify considerably if you can index
+outside the range `1:size(A,d)` (and not just `0:size(A,d)-1`,
+either).
+
+If you have cases where you think such indexing might help you,
+starting with julia-0.5 you have reasonable prospects for making this
+work. While the "unconventional" array types will be provided via
+packages, the core language has been substantially (though surely
+incompletely) generalized to ensure that algorithms work as expected
+with such arrays.
+
+The purpose of the blog post is to address the question, "what do I
+have to do to support such arrays in my own code?"  First, let's
+address the simplest case: if you know that your code will never need
+to handle arrays with unconventional indexing, hopefully the answer is
+"nothing." Old code, on conventional arrays, should function
```

Comment:
In American English, periods are always included inside the quote.  In British English, they are always outside.
http://www.thepunctuationguide.com/british-versus-american-style.html, last paragraph